### PR TITLE
Add Program tests and null guard for UseStartup

### PR DESCRIPTION
**Pull request by HaoK**
cc @davidfowl

### CODE REVIEW

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.AspNet.Hosting/HostingEngine.cs [hunk 1]
```diff
@@ -124,7 +124,7 @@ private void EnsureServer()
                 // Blow up if we don't have a server set at this point
                 if (_serverFactoryLocation == null)
                 {
-                    throw new InvalidOperationException("UseStartup() is required for Start()");
+                    throw new InvalidOperationException("UseServer() is required for Start()");
```
Comment (Eilon) [inline src/Microsoft.AspNet.Hosting/HostingEngine.cs]: Use `nameof()`?

Comment (Eilon) [inline src/Microsoft.AspNet.Hosting/HostingEngine.cs]: (Can do it with fancy C# 6 string interpolation too.)

**Comment by HaoK** [comment]
Rolled these changes in with https://github.com/aspnet/Hosting/pull/237/files, closing this PR
